controllers/openstackcluster_controller.go

```diff
@@ -103,15 +99,66 @@ func (r *OpenStackClusterReconciler) Reconcile(request ctrl.Request) (_ ctrl.Res
 
 	// Handle deleted clusters
 	if !openStackCluster.DeletionTimestamp.IsZero() {
-		return r.reconcileClusterDelete(logger, cluster, openStackCluster)
+		return reconcileDelete(log, r.Client, cluster, openStackCluster)
 	}
 
 	// Handle non-deleted clusters
-	return r.reconcileCluster(ctx, logger, patchHelper, cluster, openStackCluster)
+	return reconcileNormal(ctx, log, r.Client, patchHelper, cluster, openStackCluster)
+}
+
+func reconcileDelete(log logr.Logger, client client.Client, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) (ctrl.Result, error) {
```

Comment:
Any particular reason to change this from a method on OpenstackClusterReconciler to a separate function?

Reply:
Mostly to make it consistent with CAPA for an easier diff :). But then later on I saw that CAPA itself does it differently between cluster and machien controller and I lost track of it a bit. I rolled it back.